scripts/build_test.py

```diff
@@ -101,3 +434,163 @@ def test_get_hashes_json_file_contents(self):
+        """Test generate_build_tasks_to_build_all_files_in_directory queues up
+        the same number of build tasks as the number of files in the source
+        directory.
+        """
+        asset_hashes = build.get_file_hashes(ASSETS_DEV_DIR)
+        tasks = collections.deque()
+
+        self.assertEqual(len(tasks), 0)
+        # Build all files.
+        tasks = build.generate_build_tasks_to_build_all_files_in_directory(
+            ASSETS_DEV_DIR, ASSETS_OUT_DIR, asset_hashes)
+        total_file_count = build.get_file_count(ASSETS_DEV_DIR)
+        self.assertEqual(len(tasks), total_file_count)
+
+    def test_generate_build_tasks_to_build_files_from_filepaths(self):
+        """Test generate_build_tasks_to_build_files_from_filepaths queues up a
+        corresponding number of build tasks to the number of file changes.
+        """
+        new_filename = 'manifest.json'
+        recently_changed_filenames = [
+            os.path.join(ASSETS_DEV_DIR, new_filename)]
+        asset_hashes = build.get_file_hashes(ASSETS_DEV_DIR)
+        build_tasks = collections.deque()
+
+        self.assertEqual(len(build_tasks), 0)
+        build_tasks += build.generate_build_tasks_to_build_files_from_filepaths(
+            ASSETS_DEV_DIR, ASSETS_OUT_DIR,
+            recently_changed_filenames, asset_hashes)
+        self.assertEqual(len(build_tasks), len(recently_changed_filenames))
+
+        build_tasks.clear()
+        svg_filepaths = build.get_filepaths_by_extensions(
+            ASSETS_DEV_DIR, ('.svg',))
+        # Make sure there is at least 1 SVG file.
+        self.assertGreater(len(svg_filepaths), 0)
+
+        self.assertEqual(len(build_tasks), 0)
+        build_tasks += build.generate_build_tasks_to_build_files_from_filepaths(
+            ASSETS_DEV_DIR, ASSETS_OUT_DIR, svg_filepaths, asset_hashes)
+        self.assertEqual(len(build_tasks), len(svg_filepaths))
+
+    def test_generate_build_tasks_to_build_directory(self):
+        """Test generate_build_tasks_to_build_directory queues up a
+        corresponding number of build tasks according to the given scenario.
+        """
+        EXTENSIONS_DIRNAMES_TO_DIRPATHS = {
+            'dev_dir': EXTENSIONS_DEV_DIR,
+            'staging_dir': os.path.join(
+                TEST_DIR, 'backend_prod_files', 'extensions', ''),
+            'out_dir': os.path.join(TEST_DIR, 'build', 'extensions', '')
+        }
+        file_hashes = build.get_file_hashes(EXTENSIONS_DEV_DIR)
+        build_dir_tasks = collections.deque()
+        build_all_files_tasks = (
+            build.generate_build_tasks_to_build_all_files_in_directory(
+                EXTENSIONS_DEV_DIR, EXTENSIONS_DIRNAMES_TO_DIRPATHS['out_dir'],
+                file_hashes))
+
+        # Test for building all files when staging dir does not exist.
+        self.assertEqual(len(build_dir_tasks), 0)
+        self.assertLess(len(build_dir_tasks), len(build_all_files_tasks))
```

Comment:
Here and below, I think you can remove a bunch of these extra checks if you simply assert above that len(build_all_files_tasks) is greater than 0.

In general if you have an "equals" and a "not equal / greater than / less than" check, prefer the "equals" check -- it's stricter. Here and in the other three places below.

Reply:
Simplified assert logic.